tests/data/test_temporal_random_walker.py

```diff
@@ -17,19 +17,30 @@
 import pandas as pd
 import pytest
 import numpy as np
-import networkx as nx
 from stellargraph.data.explorer import TemporalRandomWalk
 from stellargraph.core.graph import StellarGraph
 
 
+def create_temporal_graph(nodes, edges):
+    edge_cols = ["source", "target", "weight"]
+    return StellarGraph(
+        nodes=pd.DataFrame(index=nodes), edges=pd.DataFrame(edges, columns=edge_cols),
+    )
+
+
 @pytest.fixture()
 def temporal_graph():
     nodes = [1, 2, 3, 4, 5, 6]
     edges = [(1, 2, 5), (2, 3, 2), (2, 4, 10), (4, 5, 3), (4, 6, 12)]
-    edge_cols = ["source", "target", "weight"]
-    return StellarGraph(
-        nodes=pd.DataFrame(index=nodes), edges=pd.DataFrame(edges, columns=edge_cols),
+    return create_temporal_graph(nodes, edges)
+
+
+def temporal_graph_negative_times(num_edges):
+    nodes = [1, 2, 3, 4, 5, 6]
+    edges = np.hstack(
+        [np.random.choice(nodes, size=(num_edges, 2)), -np.ones((num_edges, 1))]
```

Comment:
Is it worth having some non-negative ones here, to ensure we don't accidentally refactor to something like `np.all(times < 0)` instead of `np.any(times < 0)`/`np.where(times < 0)`? E.g. 

```suggestion
        [np.random.choice(nodes, size=(num_edges, 2)), -np.repeat([-1, 1], num_edges / 2)]
```

Plus adjusting `1` to `2` in the test parameterisation, and the `num_edges > 10` threshold too.

Reply:
I might go with `temporal_graph_negative_times(num_neg_edges)` then this becomes
```python
[np.random.choice(nodes, size(num_neg_edges * 2, 2)), -np.repeat([-1, 1], num_neg_edges)]
```

Reply:
That's better :+1: (also, I noticed that my suggestion and your replacement accidentally retain the `-` in front of the `np.repeat`, which is probably unnecessary)